delegator/prisma/schema.prisma

```diff
@@ -7,47 +7,33 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
-model Transaction {
-  id          BigInt    @id @default(autoincrement()) @map("transaction_id")
-  from        String    @db.VarChar(42)
-  to          String    @db.VarChar(42)
-  input       String    @db.VarChar(1024)
-  gas         String    @db.VarChar(20)
-  value       String    @db.VarChar(20)
-  chainId     String    @db.VarChar(20)
-  gasPrice    String    @db.VarChar(20)
-  nonce       String    @db.VarChar(20)
-  v           String    @db.VarChar(66)
-  r           String    @db.VarChar(66)
-  s           String    @db.VarChar(66)
-  rawTx       String    @db.VarChar(1024)
-  signedRawTx String?   @db.VarChar(1024)
-  succeed     Boolean?
-  function    Function? @relation(fields: [functionId], references: [id])
-  functionId  BigInt?   @map("function_id")
-  contract    Contract? @relation(fields: [contractId], references: [id])
-  contractId  BigInt?   @map("contract_id")
-  reporter    Reporter? @relation(fields: [reporterId], references: [id], onDelete: Cascade)
-  reporterId  BigInt?   @map("reporter_id")
-
-  @@map("transactions")
-}
-
 model Organization {
   id   BigInt @id @default(autoincrement()) @map("organization_id")
   name String @unique @db.VarChar(50)
 
-  Reporter Reporter[]
+  reporter Reporter[]
 
   @@map("organizations")
 }
 
+model Reporter {
+  id             BigInt        @id @default(autoincrement())
+  address        String        @unique @db.VarChar(42)
+  organization   Organization  @relation(fields: [organizationId], references: [id], onDelete: Cascade)
+  organizationId BigInt        @map("organization_id")
+  contract       Contract[]    @relation(name: "WhitelistTable")
+  transaction    Transaction[]
+
+  @@map("reporters")
+}
+
 model Contract {
-  id          BigInt        @id @default(autoincrement()) @map("contract_id")
-  address     String        @unique @db.VarChar(42)
-  Function    Function[]
-  Reporter    Reporter[]
-  Transaction Transaction[]
+  id                BigInt        @id @default(autoincrement()) @map("contract_id")
+  address           String        @unique @db.VarChar(42)
+  allowAllFunctions Boolean?      @default(false)
+  function          Function[]
+  transaction       Transaction[]
+  reporter          Reporter[]    @relation(name: "WhitelistTable")
```

Comment:
I think we do not need to name it table, and we should avoid capital letters. We could name it maybe `contract_reporter_whitelist`. What do you think?